### ISSUE TITLE
Inventory UI Rework

### ISSUE THREAD

**Issue by LexiBasilisk** [Yellow-Dog-Man/Resonite-Issues#1119]
### Is your feature request related to a problem? Please describe.

The existing inventory UI is old, clunky, and has several known issues.

### Describe the solution you'd like

Remaking the inventory UI to be something more fully-featured, and polished.

### Describe alternatives you've considered

N/A

### Additional Context

Making this issue as a 'hub issue' as a separate work item to track dependencies/blockers related to this work. E.g.:
#10
#31
#108
#289
#534
#771
#1105

**Comment by Frooxius** [comment]
Some notes on what's planned for the inventory UI rework:
- The UI will be reworked using the new data feed system, allowing you to make your own inventories or modify the UI whichever way you want easily
- Search will be natively supported
- Functionality to move/rename items (both saved items and folders)
- Ability to have multiple inventory views - e.g. placing a facet with quick access to folders on your hands or making more views and tabs
- Ability to view metadata of saved items without spawning them (e.g. amount of assets, texture data, total geometry and so on)
- Better item saving flow with visualization of items that are currently being synced
- Favorites, recent items/folders & folder hierarchy browser
- Native support for locally stored items for saving/browsing items locally

**Comment by FlameSoulis** [comment]
One thing that I'm unsure if it'd fall under this or not would be the thumbnail loading: if you have many items in a folder, the client doesn't like that and chugs when loading more than 20 item folders.

That said, having more functionality is already a major selling point of this ticket.

**Comment by Frooxius** [comment]
I'd say for now that falls under. The goal is to make the new inventory UI responsive and fast using the new systems, so we'll see how those handle sudden burst of thumbnails.

If it turns out that to make that smoother needs to be addressed by another thing - e.g. say optimizations of loading of assets and such, we'll split if off at that point.

**Comment by Frooxius** [comment]
I don't see that working well with the variety of different items that can be spawned. It doesn't match with how items are spawned in Resonite, they're not just environment props to be placed.

**Comment by lynrayy** [comment]
Then maybe make the system a little more complicated? Let the creator of the item be able to set the spawn option - place on the surface / place in the air.
You can also generate a dimension cube that will be displayed before placement. This is something like a lightweight thumbnail representing the size of the item being placed in the world.

With the "place in the air" option, you can move the item away from you by scrolling the mouse wheel before placing it. Although, moving the spawn point away can also be done when placing on the surface

**Comment by Frooxius** [comment]
The goal of the Inventory UI rework is to make the system simpler to use, not more complicated.

**Comment by lynrayy** [comment]
Simplicity and convenience do not always go hand in hand. I can call the placement of an object simple when it appears in front of the face. But is it convenient? Not very. There must be a balance between convenience and simplicity. And when it is achieved, and there is also clarity of the system, then the system becomes user friendly.

**Comment by lynrayy** [comment]
I can of course say that the placement preview is not really related to the UI, so maybe this is a separate topic for discussion

**Comment by Frooxius** [comment]
That's not what I meant. By simple, I mean system that's convenient to use.

**Comment by Frooxius** [comment]
Like I said, that'd be separate from this UI rework though.

**Comment by lynrayy** [comment]
So maybe create a new issue?

**Comment by Redd56** [comment]
yeah ima make an issue for that, an in inventory 3d view would actualy be kinda neat to have

**Comment by jvyden** [comment]
I don't really think there's too much that's worth considering from VRC's UI for this. The UI we have is already a lot more sophisticated.

I do like the idea of adding items to your context menu, perhaps that could be an official method of cloud spawning in the future.

But outside the folders being replaced with collections and a ordering option we pretty much have the same thing in terms of flow and UX... so I'm not sure what's meant to be taken from VRC, other than the spawning system mentioned above that I wouldn't really consider part of the 'UI' per-se.
